pkg/volume/glusterfs/glusterfs.go

```diff
@@ -910,6 +928,33 @@ func parseClassParameters(params map[string]string, kubeClient clientset.Interfa
 		return nil, fmt.Errorf("StorageClass for provisioner %s must contain 'resturl' parameter", glusterfsPluginName)
 	}
 
+	if len(parseVolumeType) == 0 {
+		cfg.volumeType = gapi.VolumeDurabilityInfo{Type: gapi.DurabilityReplicate, Replicate: gapi.ReplicaDurability{Replica: replicaCount}}
+	} else {
+		parseVolumeTypeInfo := dstrings.Split(parseVolumeType, ":")
+		switch parseVolumeTypeInfo[0] {
+		case "replicate":
+			replicaCount, err := convertVolumeParam(parseVolumeTypeInfo[1])
```

Comment:
you should check that `parseVolumeTypeInfo` has enough items, using `parseVolumeTypeInfo[1]` directly can crash for malformed `parseVolumeType`.